javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javaparsermodel/contexts/MethodCallExprContext.java

```diff
@@ -150,12 +152,18 @@ private MethodUsage resolveMethodTypeParametersFromExplicitList(TypeSolver typeS
     @Override
     public SymbolReference<ResolvedMethodDeclaration> solveMethod(String name, List<ResolvedType> argumentsTypes, boolean staticOnly) {
         Collection<ResolvedReferenceTypeDeclaration> rrtds = findTypeDeclarations(wrappedNode.getScope());
+
+        if (rrtds.isEmpty()) {
+            rrtds = Collections.singleton(typeSolver.solveType(Object.class.getCanonicalName()));
```

Comment:
Question: Should we manually specify `java.lang.Object` here as opposed to trying to resolve the object class?

Are there any cases where it won't resolve to `java.lang.Object`? 

I do seem to recall a discussion about supplying a custom `Object` class (perhaps in the context of Android app development and supplying a jar file which contains an Object.java???), but I'll need to have a search to figure out if it is relevant/applicable here, so my current thought is "maybe"..?

Reply:
no, you're right - `<T>` is always equivalent to `<T extends java.lang.Object>`

I'm not sure about manually specifying vs resolving - the way I wrote it seemed like the most straightforward way to obtain a `ResolvedReferenceTypeDeclaration` for `java.lang.Object`, but I'm open to better ideas.

Android still uses `java.lang.Object`, so I can't think of any "custom implementation" issues - the public API's the public API

Reply:
It still niggles at me that perhaps there is a way for the reference to Object not be `java.lang.Object` but can't yet figure out if/how this might actually be the case (e.g. an inner class somewhere, or perhaps an `Object` located somewhere other than `java.lang`). 

Personally I would be more comfortable explicitly referring to `java.lang.Object`, but this shouldn't block merging of the PR as-is though :+1: